src/test/java/com/example/joy/myFirstSpringBoot/controllers/BirthdayInfoControllerIT.java

```diff
@@ -0,0 +1,112 @@
+package com.example.joy.myFirstSpringBoot.controllers;
+
+import com.example.joy.myFirstSpringBoot.services.BasicBirthdayService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.user;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ContextConfiguration(classes = { BirthdayInfoController.class,
+								BasicBirthdayService.class})
+@AutoConfigureMockMvc
+@WebMvcTest
+class BirthdayInfoControllerIT {
```

Comment:
Renamed to *IT.java so failsafe picks it up by default (you can configure the plugin if you liked the previous name better)